src/Common/src/System/IO/StringParser.cs

```diff
@@ -171,17 +171,12 @@ public unsafe int ParseNextInt32()
                     {
                         ThrowForInvalidData();
                     }
-                    result = checked((result * 10) + d);
+                    result = negative ? checked((result * 10) - d) : checked((result * 10) + d);
```

Comment:
The previous way was throwing overflow exception as int.minvalue = -2147483648 and while getting the digits as positive there was overflow at checked((214748364 * 10) + 8) as int.maxvalue = 2147483647